func.go

```diff
@@ -15,12 +15,23 @@ type baseFuncObject struct {
 	lenProp valueProperty
 }
 
-type funcObject struct {
+type baseJsFuncObject struct {
 	baseFuncObject
 
-	stash *stash
-	prg   *Program
-	src   string
+	stash  *stash
+	prg    *Program
+	src    string
+	strict bool
+}
+
+type funcObject struct {
+	baseJsFuncObject
+}
+
+type arrowFuncObject struct {
+	baseJsFuncObject
+	this      Value
+	newTarget Value
 }
```

Comment:
[q] should we modify/add the `MemUsage` function for these objects?

Reply:
Good call, let me take a look!

Reply:
Actually I will create a separate ticket so that I can first try to pull in the remaining changes from goja.